lib/diameter-connection.js

```diff
@@ -33,10 +33,9 @@ function DiameterConnection(options, socket) {
         try {
             buffer = Buffer.concat([buffer, data instanceof Buffer ? data : new Buffer(data)]);
 
-            // If we collected header
-            if (buffer.length >= DIAMETER_MESSAGE_HEADER_LENGTH_IN_BYTES) {
+            while (buffer.length >= DIAMETER_MESSAGE_HEADER_LENGTH_IN_BYTES) {
                 var messageLength = diameterCodec.decodeMessageHeader(buffer).header.length;
-
+                
```

Comment:
Runaway spaces! :) (Try installing an EditorConfig extension to your IDE)

Reply:
I have it (vscode), but of course it decided to stop working..